framework/src/Volo.Abp.Features/Volo/Abp/Features/FeatureInterceptor.cs

```diff
@@ -29,7 +30,7 @@ public async override Task InterceptAsync(IAbpMethodInvocation invocation)
 
         protected virtual async Task CheckFeaturesAsync(IAbpMethodInvocation invocation)
         {
-            await _methodInvocationFeatureCheckerService.CheckAsync(
+            await _serviceProvider.GetRequiredService<IMethodInvocationFeatureCheckerService>().CheckAsync(
```

Comment:
What if this interceptor is applied to a singleton service. In this case, `FeatureInterceptor` instance (related to the singleton service) will also be a single instance and `IServiceProvider` will be the root service provider that is never disposed. If we resolve `IMethodInvocationFeatureCheckerService` in every method call, then it may cause to a memory leak since `IMethodInvocationFeatureCheckerService` instaces will never be disposed.
Can you think about that. You can test for example to see if I am right.

We should extremely careful when resolving from `IServiceProvider`. It can be a problem if that's not a scoped provider.

For this example, if we cache the `IMethodInvocationFeatureCheckerService` instance (like we do with the new LazyServiceProvider), then we don't have this problem. Alternatively, we may keep using the constructor injection.

Can you check this PR from this point?

Reply:
Now `IServiceScopeFactory` is used in the interceptor. Other places with HTTP context continue to use `IServiceProvider`

Reply:
I rolled back MethodInvocationAuthorizationService for the same reason: https://github.com/abpframework/abp/commit/7d049361d716c3e720ea1e801e3cce1113fb0b8d